src/server/middleware/twitter.middleware.ts

```diff
@@ -304,6 +313,172 @@ function loadTwitterData(
     });
 }
 
+function loadTwitterDataV2(
+  credentials: TwitterApiCredentials,
+  request: GetTweetsRequest
+): Promise<TwitterApiResponse> {
+  const requestUrl = 'https://api.twitter.com/2/tweets/search/all';
+  const user = request.credentials?.additionalUserInfo?.username;
+  if (!user) {
+    throw new Error('No user credentials in GetTweetsRequest');
+  }
+
+  const params = {
+    // Include next_token if it's part of the request.
+    ...(request.nextPageToken && { next_token: request.nextPageToken }),
+    ...{
+      query: `(@${user} OR url:twitter.com/${user}) -from:${user} -is:retweet`,
+      max_results: BATCH_SIZE,
+      'user.fields': 'id,name,username,profile_image_url,verified',
+      expansions: 'author_id,attachments.media_keys,referenced_tweets.id',
+      start_time: formatTimestampForV2(request.startDateTimeMs),
+      end_time: formatTimestampForV2(request.endDateTimeMs),
+      'media.fields': 'url,type',
+      'tweet.fields':
+        'attachments,created_at,id,entities,lang,public_metrics,source,text',
+    },
+  };
+
+  return axios
+    .get<TwitterApiResponse>(requestUrl, {
+      headers: {
+        authorization: `Bearer ${credentials.bearerToken}`,
+      },
+      params,
+      transformResponse: [
+        (data, _) => {
+          const parsed = JSON.parse(data);
+          const tweets: V2TweetObject[] = parsed.data ?? [];
+          const includes: V2Includes = parsed.includes ?? {};
+          return <TwitterApiResponse>{
+            results: tweets.map((tweet) => packV2Tweet(tweet, includes)),
+            next: parsed.meta.next_token,
+          };
+        },
+      ],
+    })
+    .then((response) => response.data);
+}
+
+// Packs a Tweet response object from the v2 Search API format into the
+// Enterprise Search API format.
+function packV2Tweet(tweet: V2TweetObject, includes: V2Includes): TweetObject {
```

Comment:
Nit: I think 'packV2TweetAsEnterprise' would be a tiny bit clearer, though it is a bit verbose.